common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java

```diff
@@ -330,4 +330,20 @@ public AccountRecord getAccount(@NonNull final String homeAccountId,
         return accountRecords.get(0);
     }
 
+    /**
+     * Read the metadata used for authority validation.
+     *
+     * @return Authority validation metadata string
+     */
+    public String getAuthorityValidationMetadata(@NonNull String environment) {
+        return getAccountCredentialCache().getAuthorityValidationMetadata(environment);
+    }
+
+    /**
+     * Save the metadata used for authority validation.
+     */
+    public void saveAuthorityValidationMetadata(@NonNull String environment, @NonNull String valueData) {
+        getAccountCredentialCache().saveAuthorityValidationMetadata(environment, valueData);
```

Comment:
Is there some reason to put this information in this particular XML file on Android?  Can it go in its own shared preferences instance?